src/todo-list/todo-list.ts

```diff
@@ -0,0 +1,364 @@
+      elCompletedItems,
+      elClearCompleted,
+    } = this.builder.createTodoToolboxElement();
+
+    this.layouts.activeItems = elActiveItems;
+    this.layouts.completedItems = elCompletedItems;
+    this.layouts.clearCompleted = elClearCompleted;
+    this.layouts.allItems = elAllItems;
+    this.layouts.itemCnt = elItemCnt;
+    this.layouts.buttonWrapper = wrapper;
+    this.layouts.buttonPack = buttonPack;
+
+    buttonPack.appendChild(elAllItems);
+    buttonPack.appendChild(elActiveItems);
+    buttonPack.appendChild(elCompletedItems);
+
+    wrapper.appendChild(elItemCnt);
+    wrapper.appendChild(buttonPack);
+    wrapper.appendChild(elClearCompleted);
+
+    this.layouts.root.appendChild(this.layouts.input);
+    this.layouts.root.appendChild(this.layouts.ul);
+    this.layouts.root.appendChild(this.layouts.buttonWrapper);
+    this.options.el.appendChild(this.layouts.root);
+  }
+
+  addItem() {
+    const newItem: TodoListItem = {
+      id: Date.now().toString(),
+      label: this.data.inputValue.trim(),
+      isChecked: false,
+      createDt: Date.now(),
+    };
+    this.data.originItems.push(newItem);
+    this.data.items = [...this.data.originItems];
+    this.dispatch();
+  }
+
+  dispatch(initialItems?: TodoListItem[]) {
+    if (initialItems) {
+      this.data.originItems = initialItems;
+    }
+    let items = [] as TodoListItem[];
+    switch (this.selectedBtn) {
+      case BUTTON_TYPES.ALL_ITEMS:
+      default:
+        items = [...this.data.originItems];
+        break;
+      case BUTTON_TYPES.ACTIVE_ITEMS:
+        items = [...this.data.originItems.filter((i) => !i.isChecked)];
+        break;
+      case BUTTON_TYPES.COMPLETED_ITEMS:
+        items = [...this.data.originItems.filter((i) => i.isChecked)];
+        break;
+      case BUTTON_TYPES.CLEAR_COMPLETED:
+        this.data.originItems = this.data.originItems.filter((i) => {
+          return !i.isChecked;
+        });
+        items = this.data.originItems;
+        this.selectedBtn = BUTTON_TYPES.ALL_ITEMS;
+        break;
+    }
+    this.renderItems(items);
+    this.renderItemCnt(this.data.originItems.filter((i) => !i.isChecked).length);
+    this.renderFilterButtons();
+    this.renderClearCompletedBtn(this.data.originItems.filter((i) => i.isChecked).length);
+  }
+
+  sortList(items: TodoListItem[]) {
+    if (items.length === 0) {
+      return [this.builder.createEmptyRow()];
+    }
+    if (this.options.useDnd) {
+      // d&d 사용시 정렬 x
+      return items.map((item) => this.builder.createRow(item));
+    }
+    return items
+      .sort((a, b) => b.createDt - a.createDt)
+      .sort((a, b) => Number(a.isChecked) - Number(b.isChecked))
+      .map((item) => this.builder.createRow(item));
+  }
+
+  renderItems(items: TodoListItem[]) {
+    const liArr: HTMLLIElement[] = this.sortList(items);
+    if (this.layouts.ul) {
+      this.layouts.ul.innerHTML = '';
+      liArr.forEach((li) => this.layouts.ul?.appendChild(li));
+    }
```

Comment:
```suggestion
    if (!this.layouts.ul) {
      return;
    }
```
다른 렌더 메서드들처럼 이것도 얼리리턴으로 할 수 있을듯합니다!